### PR TITLE
Add slicetrace program for producing diagnostics from traces

### PR DESCRIPTION
**Pull request by jcharum**
`slicetrace` consumes a Chrome trace event format file produced by a bigslice program and prints some useful diagnostics from it, in particular aggregating information about total compute time by (invocation, op), quartiles of task execution time, segmenting of overhead, etc.

### CODE REVIEW

**Comment by jcharum** [comment]
This depends on #55 and #57, so I'll merge them before requesting review on this one. However, I provide this to give context to #55 and #57.

**Comment by jcharum** [comment]
Example output from `slicer reduce`:
```
# inv1:summary
    location    /home/ubuntu/code/bigslice/cmd/slicer/reduce.go:46
        args                                       "64", "1000000"

# inv1:slice
            op    shards       start    duration
    reader_map        64    00:00:00    00:00:27
        reduce        64    00:00:27    00:00:05

# inv1:task:io
            op       total        read       write
    reader_map    00:04:25    00:00:00    00:00:50
        reduce    00:01:06    00:00:00    00:00:00

# inv1:task:quartile
            op         min          q1          q2          q3         max
    reader_map    00:00:03    00:00:04    00:00:04    00:00:04    00:00:05
        reduce    00:00:00    00:00:00    00:00:01    00:00:01    00:00:01
```

File: cmd/slicetrace/session.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+			q1:       q1,
+			q2:       q2,
+			q3:       q3,
+			max:      a.durations[len(a.durations)-1],
+		})
+	}
+	return opStats
+}
+
+func computeQuartiles(ds []time.Duration) (time.Duration, time.Duration, time.Duration) {
```
Comment (josh-newman) [inline cmd/slicetrace/session.go:233]: It's not very likely, but it'd be nice to not panic for [any number of tasks](https://play.golang.org/p/5nu5fh76Tdv).

File: cmd/slicetrace/session.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+	mid := len(ds) / 2
+	if len(ds)%2 == 0 {
+		return (ds[mid-1] + ds[mid]) / 2
+	} else {
+		return ds[mid]
+	}
+}
+
+func truncatef(v interface{}) string {
+	b := limitbuf.NewLogger(80)
```
[the remark this answers is not in the document]
Comment (jcharum) [reply cmd/slicetrace/session.go:259]: This is a pretty arbitrary number, as this gets printed as a column by a `TabWriter` with a preceding column. I thought I'd start with 80 and then see if people needed more to jog memories about what trace they're looking at.

[the remark this answers is not in the document]
Comment (josh-newman) [reply cmd/slicetrace/session.go:262]: This was just a joke about @jjc-coursera's editor preferences.

**Pull request merged**